src/Evolution/Actions/RandomizeInitialFields.hpp

```diff
@@ -0,0 +1,124 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <functional>
+#include <optional>
+#include <pup.h>
+#include <pup_stl.h>
+#include <random>
+#include <string>
+#include <tuple>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Options/Auto.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/PupStlCpp17.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+struct GlobalCache;
+}  // namespace Parallel
+namespace tuples {
+template <typename... Tags>
+struct TaggedTuple;
+}  // namespace tuples
+/// \endcond
+
+namespace evolution::Actions {
+
+/*!
+ * \brief Optionally add random noise to the initial guess.
+ *
+ * In an evolution, add this action to the action list after initializing the
+ * evolved fields but before taking timesteps to add random noise to the evolved
+ * fields. The random noise can be toggled and is configurable with input-file
+ * options. Adding random noise to the initial fields can be useful to test the
+ * convergence and stability of evolutions.
+ */
+template <typename System>
+struct RandomizeInitialFields {
```

Comment:
AFAICT this action differs from the one in `elliptic::Actions` in exactly two places. Should we just move the elliptic action to `ParallelAlgorithms/Actions`, call it `RandomizeVariables`, and template it on `VariablesTag` instead of `System`?

The second line that differs from the elliptic action (the first is the `VariablesTag`), is the name of the option (L76). You can supply a name by giving the action a second template parameter `Label`, and return `Options::name<Label>()` in line 76.

Pros: We save 230 lines of code to maintain. Upgrades that I planned to do, like choosing the distribution in the input file, carry over to both actions. The action can be used to randomize _any_ variables in the DataBox. Cons: Can't make evolution/elliptic-specific modifications, if any come up.

@wthrowe @geoffrey4444 what are your thoughts on this?

Reply:
I hadn't looked at the elliptic action.  Yes, let's merge them.

I suspect system-specific issues are more likely than problems with elliptic vs. evolution.  Adding random noise to the density of a bounded matter source would give unphysical data, for example, but I think we can deal with that when someone wants to use this in such a system.